tests/apply-diff.bats

```diff
@@ -44,3 +44,61 @@ load helpers
 	checkchanges
 	checkdiffs
 }
+
+@test "apply-diff-from-staging-directory" {
+	case "$STORAGE_DRIVER" in
+	overlay*)
+		;;
+	*)
+		skip "driver $STORAGE_DRIVER does not support diff-from-staging-directory"
+		;;
+	esac
+
+	SRC=$TESTDIR/source
+	mkdir -p $SRC
+	createrandom $SRC/file1
+	createrandom $SRC/file2
+	createrandom $SRC/file3
+
+	local sconf=$TESTDIR/storage.conf
+
+	local root=`storage status 2>&1 | awk '/^Root:/{print $2}'`
+	local runroot=`storage status 2>&1 | awk '/^Run Root:/{print $3}'`
+
+	cat >$sconf <<EOF
+[storage]
+driver="overlay"
+graphroot="$root"
+runroot="$runroot"
+
+[storage.options]
+pull_options = {enable_partial_images = "true" }
+EOF
+
+	# Create a layer.
+	CONTAINERS_STORAGE_CONF=$sconf run storage --debug=false create-layer
+	[ "$status" -eq 0 ]
+	[ "$output" != "" ]
+	layer="$output"
+
+	CONTAINERS_STORAGE_CONF=$sconf run storage --debug=false applydiff-using-staging-dir $layer $SRC
+        cat <<< $output
```

Comment:
Why is this different than the test at line 81?  If $output is empty, doesn't it still succeed?

Reply:
the command doesn't print anything on success.  I'll drop this line as it is not needed.